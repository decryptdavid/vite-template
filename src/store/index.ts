import { createStore } from 'vuex';

interface State {
  count: number;
}

export const store = createStore({
  state(): State {
    return {
      count: 0,
    };
  },
  mutations: {
    increment(state: State) {
      state.count++;
    },
  },
});
