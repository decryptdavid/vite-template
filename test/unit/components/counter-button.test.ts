import { test, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import CounterButton from '@/components/counter-button.vue';
import { store } from '@/store';

test('renders component', () => {
  const wrapper = shallowMount(CounterButton, {
    global: {
      provide: {
        store,
      },
    },
  });

  expect(wrapper.element).toMatchSnapshot();
});
