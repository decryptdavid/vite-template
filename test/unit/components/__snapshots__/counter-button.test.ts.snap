// Vitest Snapshot v1

exports[`renders component 1`] = `
<button
  class="inline-flex items-center justify-center px-5 py-3 text-base font-medium leading-6 text-white dark:text-slate-900 transition duration-150 ease-in-out bg-emerald-600 border border-transparent rounded-md hover:bg-emerald-500 focus:outline-none"
  type="button"
>
   count is: 0
</button>
`;
